plugins/catalog/src/alpha.ts

```diff
@@ -17,38 +17,46 @@
 export * from './translation';
 
 import {
-  DiscoveryApi,
-  FetchApi,
+  createApiFactory,
   discoveryApiRef,
   fetchApiRef,
+  storageApiRef,
 } from '@backstage/core-plugin-api';
 import { CatalogClient } from '@backstage/catalog-client';
 import {
   createApiExtension,
   createPlugin,
 } from '@backstage/frontend-plugin-api';
-import { catalogApiRef } from '@backstage/plugin-catalog-react';
+import {
+  catalogApiRef,
+  starredEntitiesApiRef,
+} from '@backstage/plugin-catalog-react';
+import { DefaultStarredEntitiesApi } from './apis';
 
 /** @alpha */
 export const CatalogApi = createApiExtension({
-  factory: {
+  factory: createApiFactory({
     api: catalogApiRef,
     deps: {
       discoveryApi: discoveryApiRef,
       fetchApi: fetchApiRef,
     },
-    factory: ({
-      discoveryApi,
-      fetchApi,
-    }: {
-      discoveryApi: DiscoveryApi;
-      fetchApi: FetchApi;
-    }) => new CatalogClient({ discoveryApi, fetchApi }),
-  },
+    factory: ({ discoveryApi, fetchApi }) =>
+      new CatalogClient({ discoveryApi, fetchApi }),
+  }),
+});
+
+/** @alpha */
+export const StarredEntitiesApi = createApiExtension({
```

Comment:
Wait, are we making these uppercase? How come? I see we have a mixture now, the `graphiqlBrowseApi` is lowercase.

Reply:
Yep we haven't settled on any particular format, it's a TODO :p